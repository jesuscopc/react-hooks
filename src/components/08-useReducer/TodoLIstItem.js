import React from 'react';

export const TodoLIstItem = ({ task, i, handleToggle, deleteTask}) => {
  return (
    <li 
      key={task}
      className="list-group-item"
    >
      <p
        className={ `${ task.done && 'complete' }`}
        onClick={ () => handleToggle(task.id)}
      >{i + 1} { task.desc }</p>
      <button 
        className="btn btn-danger"
        onClick={ () => deleteTask(task.id) }
        >
        Borrar
      </button>
    </li> 
  )
}
