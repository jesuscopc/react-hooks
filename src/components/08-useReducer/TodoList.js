import React from 'react';
import { TodoLIstItem } from './TodoLIstItem';

export const TodoList = ( {todos, handleToggle, deleteTask } ) => {
  return (
    <ol className="list-group list-group-flush">
      { 
        todos.map( (task, i) => (
          <TodoLIstItem
            key={task.id}
            task={task}
            i={i}
            handleToggle={handleToggle}
            deleteTask={deleteTask} 
          />
        ))
      }
    </ol>
  )
}
