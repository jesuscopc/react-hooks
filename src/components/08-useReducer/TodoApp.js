import React, { useEffect, useReducer } from 'react';
import { todoReducer } from './todoReducer';
import './todo.css';
import { TodoList } from './TodoList';
import { TodoAdd } from './TodoAdd';

export const TodoApp = () => {
  const init = () => {
    return JSON.parse( localStorage.getItem('todos')) || [];
  };
  
  const [ todos, dispatch ] = useReducer(todoReducer, [], init);

  useEffect(() => {
    localStorage.setItem('todos', JSON.stringify(todos));
  }, [todos])

  const deleteTask = (id) => {
    const action = {
      type: 'delete',
      payload: id
    }
    dispatch(action);
  };

  const handleToggle = ( todoId ) => {
    dispatch({
      type: 'toggle',
      payload: todoId
    })
  }

  const handleAddTodo = (newTodo) => {
    dispatch({
      type: 'add',
      payload: newTodo
    });
  }

  return (
    <div>
      <h1>TodoApp ( { todos.length } )</h1>
      <div className="row">
          <div className="col-7">
            todos
            <TodoList
              todos={todos}
              handleToggle={handleToggle}
              deleteTask={deleteTask} 
            />
          </div>
          <div className="col-5">
            <TodoAdd
              handelAddTodo={ handleAddTodo } 
            />
          </div>
      </div>
    </div>
  )
}
