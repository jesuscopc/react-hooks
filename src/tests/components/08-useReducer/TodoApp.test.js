import React from 'react';
import { demoTodos } from '../../fixtures/demoTodo';
import { shallow, mount } from 'enzyme';
import { TodoApp } from '../../../components/08-useReducer/TodoApp';
import { act } from '@testing-library/react';

describe('Sui Test <TodoApp/>', () => {
  const wrapper = shallow(<TodoApp/>);

  test('should be render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  
})
