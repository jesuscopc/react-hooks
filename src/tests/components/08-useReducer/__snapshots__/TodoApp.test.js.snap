// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui Test <TodoApp/> should be render correctly 1`] = `
<div>
  <h1>
    TodoApp ( 
    0
     )
  </h1>
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      todos
      <TodoList
        deleteTask={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <TodoAdd
        handelAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
